python/gps/algorithm/cost/ioc_layers.py

```diff
@@ -33,11 +35,14 @@ def setup(self, bottom, top):
     def reshape(self, bottom, top):
         top[0].reshape(1)
         self.num_demos = bottom[0].data.shape[0]
-        self.num_samples = bottom[0].data.shape[0]
+        self.num_samples = bottom[1].data.shape[0]
         self.T = bottom[0].data.shape[1]
 
         # TODO(kevin) construct helper numpy arrays to store demo_counts_
         # and sample_counts_  (as in ioc_loss_layer.cpp)
+        self.demo_counts_ = np.zeros((self.num_demos))
```

Comment:
In python, it's standard to use self._xxx for private variables, rather than self.xxx_ (which is more standard in C++). Let's use self._demo_counts here.